test/app-luatest/gh_7746_fiber_channel_close_test.lua

```diff
@@ -0,0 +1,62 @@
+local compat = require('tarantool').compat
+local t = require('luatest')
+local g = t.group()
+local fiber = require('fiber')
+
+g.test_channel_close_default = function()
+    -- No graceful close (old) by default.
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    t.assert_equals(ch:get(), 'one')
+    ch:close()
+    t.assert(ch:is_closed())
+    t.assert_not(ch:put('three'))
+    t.assert_equals(ch:get(), nil)
+end
+
+g.test_channel_close_old = function()
+    compat.fiber_channel_graceful_close = 'old'
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    t.assert_equals(ch:get(), 'one')
+    ch:close()
+    t.assert(ch:is_closed())
+    t.assert_not(ch:put('three'))
+    t.assert_equals(ch:get(), nil)
+end
+
+g.test_channel_close_new = function()
+    -- Graceful close (new) selected.
+    compat.fiber_channel_graceful_close = 'new'
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    ch:close()
+
+    t.assert_not(ch:is_closed())
+    t.assert_not(ch:put('three'))
+
+    t.assert_equals(ch:get(), 'one')
+    t.assert_not(ch:is_closed())
+
+    t.assert_equals(ch:get(), 'two')
+    t.assert(ch:is_closed())
+end
+
+g.test_close_graceful_on_empty = function()
+    compat.fiber_channel_graceful_close = 'new'
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:get()
+    ch:close()
+    t.assert(ch:is_closed())
+
+    -- Check explicitly true.
+    ch = fiber.channel(10)
+    ch:put('one')
+    ch:get()
+    ch:close()
+    t.assert(ch:is_closed())
```

Comment:
I see no difference from the previous hunk. What is the point here?

Reply:
A fiber or a fiber channel? TBH, I still don't understand. Can you point to those different branches in the code?

Reply:
Tipo, fiber channel. When there is a message in buffer and it is removed, the `channel_destroy()` is called from `get()` and when buffer is empty on `close()`, destroy is called from from `channel_close()`.

Reply:
I misinterpreted your question, yest the latter part of the `on_empty` test is redundant. Removed it.